### PR TITLE
Add a default generator

### PR DESCRIPTION
**Pull request by danieldk**
## Description

`DefaultGenerator` can be used when a generation model does not require custom prompting or other model-specific settings such as a non-standard end-of-sequence identifier.

This change reimplements `DollyV2Generator` and `FalconGenerator` using `DefaultGenerator`.

The `AutoGenerator` class it moved to the `generation` package to avoid import cycles.

### Types of change

Feature

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: curated_transformers/generation/auto_generator.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+
+class AutoGenerator(AutoModel[GeneratorWrapper]):
+    """Causal LM generator loaded from the Hugging Face Model Hub.
+
+    **NOTE** - This class can currently only be used with the following models:
+        - Models based on Dolly v2 (contain ``dolly-v2``` in the name).
+        - Models based on Falcon (contain ``falcon`` in the name).
+        - Models based on LLaMA (contain ``llama`` in the name).
+    """
```
Comment (shadeMe) [inline curated_transformers/generation/auto_generator.py:30]:
```suggestion
    """
     .. highlights::
    	Causal LM generator loaded from the Hugging Face Model Hub.

        .. attention::
          	This class can currently only be used with the following models:
          	 - Models based on Dolly v2 (contain ``dolly-v2`` in the name).
          	 - Models based on Falcon (contain ``falcon`` in the name).
          	 - Models based on LLaMA (contain ``llama`` in the name).
    """
```
To keep in line with the upcoming docs PR.

File: curated_transformers/generation/default_generator.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+from .string_generator import StringGenerator
+
+# Only provided as typing.Self in Python 3.11+.
+Self = TypeVar("Self", bound="DefaultGenerator")
+
+
+class DefaultGenerator(GeneratorWrapper, FromHFHub):
+    """
+    Generator wrapper for models that do not need specific prompting.
+    """
```
Comment (shadeMe) [inline curated_transformers/generation/default_generator.py:25]:
```suggestion
    """
    .. highlights::
    	Generator wrapper for models that do not need specific prompting.
    """
```

File: curated_transformers/generation/auto_generator.py (hunk 1 above)
Comment (danieldk) [reply curated_transformers/generation/auto_generator.py:30]: I am not sure I understand these? `highlights` is for block quotes?

https://docutils.sourceforge.io/docs/ref/rst/directives.html#highlights

Comment (shadeMe) [reply curated_transformers/generation/auto_generator.py:30]: As discussed elsewhere, we can skip the `.. highlights::` annotation in favour of a global Sphix config setting.

Comment (danieldk) [reply curated_transformers/generation/auto_generator.py]: Added the attention block.

**Pull request merged**